services/tables/src/main/java/com/linkedin/openhouse/tables/utils/CronScheduleGenerator.java

```diff
@@ -0,0 +1,72 @@
+package com.linkedin.openhouse.tables.utils;
+
+import com.cronutils.builder.CronBuilder;
+import com.cronutils.model.CronType;
+import com.cronutils.model.definition.CronDefinitionBuilder;
+import com.cronutils.model.field.expression.FieldExpression;
+import com.cronutils.model.field.expression.FieldExpressionFactory;
+import com.linkedin.openhouse.common.exception.RequestValidationFailureException;
+import com.linkedin.openhouse.tables.api.spec.v0.request.components.ReplicationConfig;
+import java.util.Random;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+/** Utility class for generating a cron schedule given an interval for which a job should run */
+@Slf4j
+@Component
+public class CronScheduleGenerator {
+
+  /**
+   * Public api to generate a cron schedule for a {@link ReplicationConfig} based on a given
+   * interval string in the form 12H, 1D, 2D, 3D.
+   *
+   * @param interval
+   * @return schedule
+   */
+  public static String buildCronExpression(String interval) {
```

Comment:
Let's call it `intervalToCronConverter` to reflect intent.

Reply:
I think adding interval in the method would be redundant as the method accepts interval. I would suggest changing to `convertToCronExpression` or `generateCronExpression`. If needed class name can be updated to `IntervalToCronConverter`.

Reply:
updated the class name to `IntervalToCronConverter` and method name to `generateCronExpression` for more clarity